src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -752,35 +779,66 @@ protected bool TakeStartLine(SocketInput input)
                 if (chFound == '?')
                 {
                     begin = scan;
-                    if (scan.Seek(ref _vectorSpaces) != ' ')
+                    if (scan.Seek(ref _vectorSpaces) == -1)
                     {
-                        return false;
+                        return RequestLineStatus.TargetIncomplete;
                     }
                     queryString = begin.GetAsciiString(scan);
                 }
 
+                if (pathBegin.GetLength(scan) == 0)
```

Comment:
I don't like calling `GetLength` this in the normal start line parsing path. I think we should use BenchmarkDotNet to compare this code path before and after for common start lines.

This is happening a lot, and we've already iterated through all this to get scan variable. I think it might be better to just make this an out param to Seek. I know we do this every time we call GetAsciiString, so maybe that should take a length parameter too.